File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/AssociatedLeftShiftRegister.cs
using System;

namespace DFIR_Compiler
{
    public class AssociatedLeftShiftRegister
    {
        private int nodeId;
        private int parentId;
        
        public AssociatedLeftShiftRegister(int NodeId, int ParentId)
        {
            parentId = ParentId;
            nodeId = NodeId;
        }

        public virtual int GetNodeId()
        {
            return nodeId;
        }
        public virtual int GetParentId()
        {
            return parentId;
        }

        public virtual void Print()
        {
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
        }

    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/AssociatedLeftShiftRegister.java
package Dfir_representation;

public class AssociatedLeftShiftRegister{
	   
	   private int parentId;
	   private int nodeId;
	   public AssociatedLeftShiftRegister(int NodeId,int parentId){
		   this.parentId = parentId;
		   this.nodeId = NodeId;
	   }
	   public int getParentId(){
		   return this.parentId;
	   }
	   public int getNodeId(){
		   return this.nodeId;
	   }
	   public void print(){
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
	   }
}